### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by SimonLeclere** [kolking/react-native-parallax-swiper#6]
I followed the documentation. For two views the animations are fluid, but as soon as you go to 3 or plus (5 in my case) the movements are jerky.

My code (if it helps) :

```js
<Swiper current={current} onChange={setCurrent} onScroll={handleScroll}>
        {views
        .map(({ title, text, images }, index) => (
          <SwiperView
            key={index}
            index={index}
            style={styles.view}
            images={images}
          >
            {/* Gradient on the content to make the bonus description readable */}
            <LinearGradient
              colors={["transparent", "black"]}
              style={StyleSheet.absoluteFillObject}
            />
            <View style={styles.content}>
              <Text style={styles.title}>{title}</Text>
              <Text style={styles.text}>{text}</Text>
            </View>
          </SwiperView>
        ))}
      </Swiper>
```

The images are all 1440x1920 px. Max 200 Ko each.

**Comment by kolking** [comment]
Hi Simon! It's hard to say what could be wrong by the code provided, it looks totally fine to me. I've tried to increase the number of views in my example to six where each view contains 5 images (2580x2796) and didn't notice any issues.

Could you try to remove the `LinearGradient` and test a production (not a debug) build?

**Comment by SimonLeclere** [comment]
Hey, thanks for your quick reply!
I'll try to remove all the superfluous elements (the gradient and other elements I didn't include in my code snippet) and make a build prod

**Comment by SimonLeclere** [comment]
Okay, for some reason, it seems like one of the elements in my component is causing these lag spikes. I'm going to try uncommenting them one by one to figure out which one. Thanks again for your help!

**Comment by SimonLeclere** [comment]
Oh, and also, it seems that when you swipe too quickly (when you swipe before the previous animation has finished), the transition becomes a bit choppy. The FPS doesn't drop though, so I don't think it's a performance issue.

It doesn't bother me for my app, but it might annoy others

**Comment by kolking** [comment]
You're welcome! I'll leave myself a note regarding the choppy transition to look into it later. Thank you for using the package in your app!

**Comment by SimonLeclere** [comment]
I did a few more tests, and I'm a bit lost :/
The app runs very smoothly using expo go, but as soon as I do a build, the fps drop (from ~80fps to ~13fps) on this screen only

**Comment by SimonLeclere** [comment]
Also, when I test with a build, some images are not rendered

**Comment by kolking** [comment]
Sounds weird to me, especially the part that _some_ images aren't rendered at all. Are you using local assets or network assets? Also, you've mentioned Expo Go, are you using their managed or bare workflow for your app? Although I didn't test in Expo, it should work just fine with bare workflow, but you have to manually install Reanimated v3 first.

**Comment by SimonLeclere** [comment]
I use the managed workflow and local assets

**Comment by kolking** [comment]
Well, I'm afraid I cannot help much with the managed workflow. It's really hard to debug when something goes wrong. Can you try a simple thing - create a new app (not Expo, just a plain RN) and implement only the swiper thing? Will it work fine or still not?

**Comment by SimonLeclere** [comment]
Apparently fps drops when several different images are loaded into memory. If I display 5 SwiperViews each with 5 images (the same ones), it doesn't lag, but if I put different images each time it lags a lot

**Comment by SimonLeclere** [comment]
https://github.com/user-attachments/assets/5e875844-6123-4dd8-bf71-5330b5fdb24d

https://github.com/user-attachments/assets/ec3d802b-526c-4cc3-971a-d1f680f8dc8d

Here are two examples:
- On the one with 5 times the same images, the animations are fluid
- On the one with different images, the FPS are very low and two layers of the last screen are not rendered

**Comment by kolking** [comment]
I don't think there's a limit. Also, in a dev build image require works slower, while in a production build it should work faster. Have you tried a production build?

**Comment by SimonLeclere** [comment]
Yes, I tried it with a production build, and the animations are indeed a little smoother. But it's still very laggy compared to expo go

**Comment by SimonLeclere** [comment]
Can you try your example but with 5 views and different images in each view (make copies of paris and sydney for example) ?

**Comment by kolking** [comment]
Could you share all the images you're using, so I can test with your images?

**Comment by SimonLeclere** [comment]
No problem, here's a we-transfer link: https://we.tl/t-Yhkhrsav2e 
Images are numbered from 1 to 4

**Comment by kolking** [comment]
Btw, this is how my example works on a dev build with 9 views:

https://github.com/user-attachments/assets/35ffd1e5-2804-4145-9f08-b13adc1263b2

**Comment by SimonLeclere** [comment]
> Btw, this is how my example works on a dev build with 9 views:
> test.mp4

Is it 9 views with different images (i mean litterally different files) ?

**Comment by kolking** [comment]
Yes, I made 3 copies of the assets folder, so each image has it's own path. And this is how it work with your images:

https://github.com/user-attachments/assets/5473753b-0e78-4731-a250-4e1091bb0485

**Comment by SimonLeclere** [comment]
Wow, it runs really smoothly, it's strange that this isn't the case for me 😭
I guess it's from expo, then :/

**Comment by kolking** [comment]
Alternatively, there could be something in your code that makes the app to re-render. Can it be the stars counter you've added? It's just hard to guess without looking into code. Have you tried to create a blank project and implement only the swiper with your images? Or you can simply take my `example` app from this repo.

**Comment by SimonLeclere** [comment]
Yes, I just tried a blank project using expo and the first example from your readme (so not even the page indicator) and it was still laggy.
Now I'm going to try it with bare workflow

**Comment by SimonLeclere** [comment]
Still laggy, it must definitely come from expo

**Comment by kolking** [comment]
Weird... I don't think that Expo is such an evil. Could you create a repo with this blank expo project and share with me?

**Comment by SimonLeclere** [comment]
Okay no problem

**Comment by SimonLeclere** [comment]
Here's the repo link: https://github.com/SimonLeclere/swiper-test

I have, in this order:
- Created an empty repo with `npx create-expo-app@latest --template blank`
- Installed react-native-parallax-swiper with `npm install @kolking/react-native-parallax-swiper`
- Installed react-native-reanimated with `npx expo install react-native-reanimated`
- Copied and pasted the [basic example from the readme](https://github.com/kolking/react-native-parallax-swiper?tab=readme-ov-file#basic-example) into App.js and imported my images.

To build the app, I use [eas](https://docs.expo.dev/tutorial/eas/configure-development-build/)

Using expo go, i have around 30fps during the animations

I'll test with a production build and edit my message

**Comment by SimonLeclere** [comment]
Okay, thanks a lot for your help!

**Comment by kolking** [comment]
Well, still works fine for me. The only difference is that I built using `npx expo run:ios`

https://github.com/user-attachments/assets/ae864b81-32a4-4b94-9325-f406a7b27fa3

**Comment by SimonLeclere** [comment]
Are you testing on an emulator or a real device ?

**Comment by kolking** [comment]
Both. The video has been taken on a real device.

**Comment by SimonLeclere** [comment]
The only difference is that I'm testing on android and you're testing on ios then. I don't have an iphone to test but I'll try to find one tomorrow. It really is a strange problem ://

**Comment by SimonLeclere** [comment]
Hello, do you have an android phone to test? I really don't understand where these lags come from

**Comment by SimonLeclere** [comment]
Hello, I finally "solved" my problem by modifying your module a little. I've replaced the ScrollView with a FlatList, which makes it possible to unload elements that are outside the view. It causes a "fade in" animation when loading images when swiping, but in my case my images are mostly transparent so it's not noticeable and it's not annoying for the user.


https://github.com/user-attachments/assets/5242b832-4528-4c3d-81ee-8452436c7bdd

I'm not entirely satisfied because I still don't know what the problem was, but at least I can move on now !

**Comment by kolking** [comment]
Great! I will probably try to replace `ScrollView` with `FlatList` to see how it works. Thanks for your persistence!
